src/Animations/babylon.animatable.ts

```diff
@@ -119,7 +119,7 @@
                         numberOfAnimationsStopped++;
                     }
 
-                    if (animations.length == numberOfAnimationsStopped) {

```

Comment:
it has no means...
animations length x,  if numberOfAnimationsStopped = y, see (animations.splice(index, 1);) then animations.length = x-y;
the check animations.length == numberOfAnimationsStopped means (x-y == y).